config/always-compiler-options.m4

```diff
@@ -268,3 +289,33 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_CC_NO_IPA_SRA], [
 	CFLAGS="$saved_flags"
 	AC_SUBST([NO_IPA_SRA])
 ])
+
+dnl #
+dnl # Check if kernel cc supports -fno-ipa-sra option.
+dnl #
+AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_KERNEL_CC_NO_IPA_SRA], [
```

Comment:
We already have a check against the CC in ZFS_AC_CONFIG_ALWAYS_CC_NO_IPA_SRA. If we are going to rewrite this to be specific to the kernel CC, we should remove the old one that nothing uses anymore.

Reply:
Separate checks for `CC` and `KERNEL_CC` are needed as those might be different compilers.
https://github.com/openzfs/zfs/blob/master/config/zfs-build.m4#L81-L88

Reply:
If we do not use the result for CC, we do not need the check.

Reply:
We do, it's injected at line 82 above to DEBUG_CFLAGS.

Reply:
My mistake. This is fine then.